src/utils/find-files.test.ts

```diff
@@ -0,0 +1,103 @@
+jest.mock('glob')
+jest.mock('path')
+jest.mock('./read-file')
+import glob from 'glob'
+import path from 'path'
+import { getFileExtensionPattern } from './read-file'
+const mockedGlob = glob as jest.Mocked<typeof glob>
+const mockedPath = path as jest.Mocked<typeof path>
+const mockedGetFileExtensionPattern = getFileExtensionPattern as jest.MockedFunction<
+  typeof getFileExtensionPattern
+>
+
+const mockedFileNames = ['/dev/config/development.yml', 'schema.json']
+const mockedResolvedPath = '/dev'
+mockedGlob.sync = jest.fn().mockReturnValue(mockedFileNames)
+mockedPath.resolve = jest.fn().mockReturnValue(mockedResolvedPath)
+mockedGetFileExtensionPattern.mockReturnValue('json')
+
+import { findConfigFiles, findFiles, isNotSchema } from './find-files'
+
+describe('findFiles works as expected', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('invokes glob.sync', () => {
+    findFiles('some/path')
+
+    expect(mockedGlob.sync).toHaveBeenCalledTimes(1)
+  })
+
+  it('returns result of glob.sync', () => {
```

Comment:
generally you should aim to make your unit tests less implementation specific and more results-oriented. what we're interested in is mainly that our code delivers the correct result, not all the invocations that it does to deliver it.

IMO a lot of the below tests could be replaced with two simpler tests:

```js
  it('returns all config files in a given directory')
```

```js
  it('filters out the schema file')
```

Reply:
Hm yeah will try to pay some attention.

But we reached a good level of nit now. For me, it's 99% about 1. having tests, 2. testing the right thing, 3. easy to maintain/update

Reply:
Agree with all of those. My point is, if you couple your tests super tightly to the implementation they are not all testing the right thing (which is mostly that the code delivers the correct result) and become harder to maintain because changing the implementation becomes more costly because you have to update more tests.

Will resolve this conversation now because refactoring all tests would be a bit much. For the next tests, tho, I'd like to challenge you to try out a little bit of TDD and writing some of your tests first, so you experience the difference.